src/msg_recorder/recorder.cc

```diff
@@ -182,7 +183,17 @@ std::string MessageRecorder::RecordDirNameGenerator() {
 }
 
 std::string MessageRecorder::SnapshotDirNameGenerator() {
-    return fmt::format("{:%Y%m%d-%H%M%S.%Z}", std::chrono::system_clock::now());
+    using print_duration_t            = std::chrono::nanoseconds;
```

Comment:
You are using nanoseconds instead of microseconds. So either update the PR and commit title, or change the code